flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint.channel;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.runtime.io.network.buffer.Buffer;
+import org.apache.flink.runtime.state.StateObject;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
+/**
+ * Writes channel state during checkpoint/savepoint.
+ */
+@Internal
+public interface ChannelStateWriter extends AutoCloseable {
+
+	/**
+	 * Initiate write of channel state for the given checkpoint id.
+	 */
+	void start(long checkpointId);
+
+	/**
+	 * Add in-flight buffers from the {@link org.apache.flink.runtime.io.network.partition.consumer.InputChannel InputChannel}.
+	 * <p>Must be called after {@link #start} and before {@link #finish(long)}.</p>
+	 * @param startSeqNum is the sequence number of the 1st passed buffer
+	 *                    It’s intended use is incremental snapshots
+	 *                    For the restored buffers it should be -1
+	 *                    For MVP it can always be -2
```

Comment:
I'd probably not refer to MVP in the code. Noone from outside will know what it means. 
How about `Can be -2 if unknown, which will fail on incremental checkpoint`.